parodos-model-api/src/main/java/com/redhat/parodos/workflow/task/shell/JiraTask.java

```diff
@@ -0,0 +1,80 @@
+package com.redhat.parodos.workflow.task.shell;
+
+import com.redhat.parodos.workflow.task.BaseWorkFlowTask;
+import com.redhat.parodos.workflows.work.WorkContext;
+import com.redhat.parodos.workflows.work.WorkReport;
+import com.redhat.parodos.workflows.work.WorkStatus;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.*;
+import java.util.function.Consumer;
+
+/**
+ * JiraTask can create/retrieve/update/delete a jira task.
+ * It returns an extended work report containing a jira ticket, and/or the api respond
+ * Any 2xx return code is COMPLETED, and anything else is FAILED
+ */
+@Slf4j
+public class JiraTask extends BaseWorkFlowTask {
+
+	public static final int DEFAULT_EXECUTION_TIMEOUT_IN_SECONDS = 600;
+
+	Consumer<String> outputConsumer = log::info;
+
+	private static class JiraTaskParams {
+
+		String id;
+		String title;
+
+		String status;
+
+		String comment;
+
+		int timeoutInSeconds = DEFAULT_EXECUTION_TIMEOUT_IN_SECONDS;
+
+		JiraTaskParams(WorkContext workContext) {
+			this.id = (String) workContext.get("id");
```

Comment:
the parameter names might be too generic in a context that may be shared among several tasks.
assuming I'm not mistaken here :-)